java-backend/src/main/java/org/kframework/backend/java/symbolic/InitializeRewriter.java

```diff
@@ -351,6 +351,12 @@ public ProcessProofRules invoke(GlobalContext rewritingContext, long initCounter
                 termContext.setTopConstraint(constraint);
                 //simplify the constraint in its own context, to force full evaluation of terms.
                 constraint = constraint.simplify(termContext);
+                if (constraint.isFalseExtended()) {
+                    StringBuilder sb = new StringBuilder();
+                    sb.append("Rule requires clause evaluates to false:\n");
+                    RuleSourceUtil.appendRuleAndSource(rule, sb);
+                    throw KEMException.criticalError(sb.toString());
+                }
```

Comment:
This is orthogonal to the `#getKLabelString` fix, right?

Reply:
It is made possible by `#getKLabelString` and is also a safety measure to avoid this sort of issue in future with builtin functions.